beacon-chain/db/filesystem/blob.go

```diff
@@ -241,22 +241,26 @@ func (bs *BlobStorage) Prune(pruneBefore primitives.Slot) error {
 
 	log.Debug("Pruning old blobs")
 
-	folders, err := afero.ReadDir(bs.fs, ".")
-	if err != nil {
-		return err
-	}
 	var totalPruned int
-	for _, folder := range folders {
-		if folder.IsDir() {
-			num, err := bs.processFolder(folder, pruneBefore)
+	err := afero.Walk(bs.fs, ".", func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() && path != "." {
+			num, err := bs.processFolder(info, pruneBefore)
 			if err != nil {
 				return err
 			}
 			blobsPrunedCounter.Add(float64(num))
 			blobsTotalGauge.Add(-float64(num))
 			totalPruned += num
 		}
+		return nil
+	})
+	if err != nil {
+		return err
 	}
+
 	pruneTime := time.Since(t)
 
 	log.WithFields(log.Fields{
```

Comment:
Can we return err at the end of this function so that we still get the log showing the number of blobs pruned (before the error)?